pulsar-testclient/src/test/java/org/apache/pulsar/testclient/PerformanceTransactionTest.java

```diff
@@ -0,0 +1,225 @@
+
+        CountDownLatch countDownLatch = new CountDownLatch(500);
+        for (int i = 0; i < 510
+                ; i++) {
+            produceToConsumeTopic.newMessage().value(("testConsume " + i).getBytes()).sendAsync().thenRun(() -> {
+                countDownLatch.countDown();
+            });
+        }
+
+        countDownLatch.await();
+
+        Thread thread = new Thread(() -> {
+            try {
+                PerformanceTransaction.main(args.split(" "));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+        thread.start();
+        thread.join();
+        Consumer<byte[]> consumeFromConsumeTopic = pulsarClient.newConsumer(Schema.BYTES)
+                .consumerName("perf-transaction-consumeVerify")
+                .topic(testConsumeTopic)
+                .subscriptionType(SubscriptionType.Shared)
+                .subscriptionName(testSub)
+                .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+                .subscribe();
+        Consumer<byte[]> consumeFromProduceTopic = pulsarClient.newConsumer(Schema.BYTES)
+                .consumerName("perf-transaction-produceVerify")
+                .topic(testProduceTopic)
+                .subscriptionName(testSub)
+                .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+                .subscribe();
+        for (int i = 0; i < 5; i++) {
+            Message message = consumeFromProduceTopic.receive(2, TimeUnit.SECONDS);
+            Assert.assertNotNull(message);
+        }
+        boolean noMessage = false;
+        for (int i = 0; i < 10; i++) {
+           Message message = consumeFromConsumeTopic.receive(2, TimeUnit.SECONDS);
+           if(message == null){
+               noMessage = true;
+           }
+        }
+        Assert.assertTrue(noMessage);
+    }
+
+
+    @Test
+    public void testProduceTxnMessage() throws InterruptedException, PulsarClientException {
+        String argString = "%s -r 10 -u %s -m %d -txn";
+        String topic = testTopic + UUID.randomUUID();
+        int totalMessage = 500;
+        String args = String.format(argString, topic, pulsar.getBrokerServiceUrl(), totalMessage);
+
+        Thread thread = new Thread(() -> {
+            try {
+                log.info("");
+                PerformanceProducer.main(args.split(" "));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+        thread.start();
+        thread.join();
+        Consumer<byte[]> consumer = pulsarClient.newConsumer().subscriptionName("subName").topic(topic)
+                .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+                .subscriptionType(SubscriptionType.Exclusive)
+                .enableBatchIndexAcknowledgment(false)
+                .subscribe();
+        for (int i = 0; i < totalMessage; i++) {
+           Message message = consumer.receive();
```

Comment:
ack this message. If ack 500 messages, we can't receive message anymore. I suggest the test message change to 100.